Add lib paths on Apple M1 (arm64)

bucanero opened the pull request:
This PR updates the checks to build the ps3 toolchain on macOS w/ Apple M1 processors.

Note: Homebrew package manager install libs on `/opt/homebrew/` when running on Apple M1 (arm64)

zeldin commented:
Hm, I'm thinking maybe instead of adding more OS-specific paths to this list, we should just check with `pkg-config` as well?

```
mikan:~% uname -a
Darwin mikan.mc.pp.se 20.6.0 Darwin Kernel Version 20.6.0: Mon Aug 30 06:12:20 PDT 2021; root:xnu-7195.141.6~3/RELEASE_ARM64_T8101 arm64
mikan:~% pkg-config --exists gmp && echo "yes"
yes
mikan:~% pkg-config --exists libelf && echo "yes"
yes
mikan:~% pkg-config --exists ncurses && echo "yes"
yes
mikan:~% pkg-config --exists zlib && echo "yes" 
yes
mikan:~% 
```

bucanero commented:
> Hm, I'm thinking maybe instead of adding more OS-specific paths to this list, we should just check with `pkg-config` as well?
> 

For sure, it looks like a more elegant (and future-proof) solution. I'm not an expert on scripting but I can try to upgrade these scripts to `pkg-config` and update the PR.

bucanero commented:
I have upgraded the scripts using `pkg-config`. Please take a look when you have a chance.

bucanero commented:
> 1. To make sure there are no regressions, I wouldn't remove the old tests, just add the pkg-config check instead of adding a new path test for homebrew

I updated the scripts, keeping the old tests, and adding the `pkg-config` check.

Btw, I added some notes about building the ps3 toolchain on M1 in #114 . I'm not sure how to include those fixes properly.

zeldin commented:
LGTM.
